string.c

```diff
@@ -6595,10 +6607,13 @@ rb_str_bytesplice(int argc, VALUE *argv, VALUE str)
     }
     str_check_beg_len(str, &beg, &len);
     str_check_beg_len(val, &vbeg, &vlen);
-    enc = rb_enc_check(str, val);
     str_modify_keep_cr(str);
+
+    if (RB_UNLIKELY(ENCODING_GET_INLINED(str) != ENCODING_GET_INLINED(val))) {
```

Comment:
There's a semantic difference here, although I'm not sure if it really matters. Previously, if the encodings were incompatible the receiver was untouched. Now `str_modify_keep_cr` is being called in all cases which would reify shared strings.

Reply:
Right. The worst case scenario is the reciever  is made independent but is unmodified from a user point of view. I think it's acceptable.